activerecord/lib/active_record/errors.rb

```diff
@@ -353,20 +353,24 @@ class Deadlocked < TransactionRollbackError
   class IrreversibleOrderError < ActiveRecordError
   end
 
+  # Superclass for all timeout-related errors
+  class Timeout < StatementInvalid
```

Comment:
Btw, this should be `TimeoutError`, but if `QueryCanceled` needs to have the same base class we should rename to `QueryExecutionInterrupted` maybe.

Reply:
> this should be TimeoutError

This confused me because most errors in `ActiveRecord` module don't have "Error" in their class: StatementInvalid, QueryCanceled, RecordNotFound, etc

Reply:
I agree with this point, maybe this could be `QueryAborted` or `QueryAbortedError`.

Reply:
`QueryAborted` is good.